### DISCUSSION TITLE
Using OpenTelemetry in the application - is it possible at all?

### DISCUSSION THREAD

**Discussion by kameeloo** [linkerd/linkerd2#9484] [Q&A]
Hi,
On the distributed-tracing topic in linkerd documentation there is a information that "Unfortunately, OpenTelemetry is not yet production ready and so OpenCensus remains our recommendation for the moment"
Does it mean OpenTelemetry is not recommended but there is a possibility to use it on my own risk?  If YES then where can I find a proper example of configuration that should be used?
Right now I have many services using OpenTelemetry and I would want to 'mesh' them with linkerd but it looks like proxy sidecar requires b3 format or OpenCensus only.

Thank you for any answer and help here,
Kind Regards,
Kamil

**Comment by adleong** [comment]
Hi @kameeloo 

The good news (ish) is that those docs are a bit out of date and the `linkerd-jaeger` plugin now uses OpenTelemetry instead of OpenCensus. 🎉 

With that said, the _propagation format_ Linkerd uses is b3, which IS compatible with OpenTelemetry: https://opentelemetry.io/docs/reference/specification/context/api-propagators/#b3-requirements

This means that in order for your application and Linkerd to participate in traces together, your application will need to use the b3 propagation format.
